src/main/java/org/candlepin/subscriptions/util/ProductOfferingSubscriptionService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.util;
+
+import com.redhat.swatch.configuration.registry.SubscriptionDefinition;
+import jakarta.ws.rs.core.Response;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.db.OfferingRepository;
+import org.candlepin.subscriptions.db.model.Offering;
+import org.candlepin.subscriptions.exception.ErrorCode;
+import org.candlepin.subscriptions.exception.MissingOfferingException;
+import org.candlepin.subscriptions.utilization.admin.api.model.OfferingProductTags;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Common methods to identify product tags between subscription and offering products */
+@Component
+@Slf4j
+public class ProductOfferingSubscriptionService {
```

Comment:
Yea I didn't find a better name since this functionality will be used by subscription and offering both. The name you suggested is good so, will change it.

Reply:
Done